File: Seccion 12 - Punteros en C++/Seccion_12_Python/05_ejercicio_05.py

nElementos: int
elementos = []

def pedir_datos():
    global nElementos, elementos
    
    nElementos = int(input('Digite el número de elementos: '))
    
    for i in range(nElementos):
        elementos.append(int(input('Digite un número: ')))

def ordenar_arreglo():
    global elementos
    aux: int
    
    for i in range(nElementos):
        for j in range(nElementos - 1):
            if elementos[j] > elementos[j+1]:
                aux = elementos[j]
                elementos[j] = elementos[j+1]
                elementos[j+1] = aux

def mostrar_arreglo():
    print('\n\nMostrando arreglo ordenado:\n')
    
    for i in range(nElementos):
        print('{}, '.format(elementos[i]), end='')
    
    print()

def main():
    
    pedir_datos()
    
    ordenar_arreglo()
    
    mostrar_arreglo()

if __name__ == '__main__':
    main()

File: Seccion 12 - Punteros en C++/Seccion_12_Dart/05_ejercicio_05.dart

import 'dart:io';

int nElementos;
List<int> elementos = new List<int>();

void pedirDatos() {
  stdout.write('Digite el número de elementos: ');
  nElementos = int.parse(stdin.readLineSync());

  for (int i = 0; i < nElementos; i++) {
    stdout.write('Digite un número[$i]: ');
    elementos.insert(i, int.parse(stdin.readLineSync()));
  }
}

void ordenarArreglo() {
  int aux;

  // Ordenar el arreglo con el método burbuja
  for (int i = 0; i < nElementos; i++) {
    for (int j = 0; j < nElementos-1; j++) {
      if (elementos[j] > elementos[j+1]) {
        aux = elementos[j];
        elementos[j] = elementos[j+1];
        elementos[j+1] = aux;
      }
    }
  }
}

void mostrarArreglo() {
  print('\n\nMostrando arreglo ordenado:\n');

  for (int i = 0; i < nElementos; i++) {
    stdout.write('${elementos[i]}, ');
  }

  print('');
}

void main() {

  pedirDatos();

  ordenarArreglo();

  mostrarArreglo();

}